travisci/python-linter_harness.sh

```diff
@@ -15,29 +15,70 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-PYTHON_SOURCE_LOCATIONS=('scripts' 'src/python')
-
 # Setup the environment variables
 # shellcheck disable=SC2155
 export PYTHONPATH=$PYTHONPATH:$(python -c 'import sysconfig; print(sysconfig.get_paths()["purelib"])')
 # more info: https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules
 export MYPYPATH=$MYPYPATH:src/python/lib
 
-PYLINT_OUTPUT_FILE=$(mktemp)
-PYLINT_ERRORS=$(mktemp)
-# CITest project is on hold and it needs to be updated before resuming its linter checker
-find "${PYTHON_SOURCE_LOCATIONS[@]}" -type f -name "*.py" \! -name "Ortheus.py" \! -name "*citest*.py" \! -path "*/citest/*" -print0 | xargs -0 pylint --rcfile pyproject.toml --verbose | tee "$PYLINT_OUTPUT_FILE"
-grep -v "\-\-\-\-\-\-\-\-\-" "$PYLINT_OUTPUT_FILE" | grep -v "Your code has been rated" | grep -v "\n\n" | sed '/^$/d' > "$PYLINT_ERRORS"
-! [ -s "$PYLINT_ERRORS" ]
+# Function to run pylint
+run_pylint() {
+  local pylint_output_file=$(mktemp)
+  local pylint_errors=$(mktemp)
+
+  # Run pylint, excluding specific files and directories
+  find "${PYTHON_SOURCE_LOCATIONS[@]}" -type f -name "*.py" \
+    ! -name "Ortheus.py" \
+    ! -name "*citest*.py" \
+    ! -path "*/citest/*" -print0 |
+    xargs -0 pylint --rcfile=pyproject.toml --verbose \
+      --msg-template='{path}:{line}:{column}: {msg_id}: {msg} ({symbol})' |
+    tee "$pylint_output_file"
+
+  # Keep only lines with pylint messages
+  grep -E '^.+:[0-9]+:[0-9]+: [A-Z]+[0-9]+: .+ (\(.*\))?$' "$pylint_output_file" >"$pylint_errors"
+
+  # Return 1 if errors were found, otherwise 0
+  local result=$(
+    ! [ -s "$pylint_errors" ]
+    echo $?
+  )
+
+  # Cleanup
+  rm "$pylint_output_file" "$pylint_errors"
+
+  return "$result"
+}
+
+# Function to run mypy, excluding certain files and paths, and capturing the outcome
+run_mypy() {
+  find "${PYTHON_SOURCE_LOCATIONS[@]}" -type f -name "*.py" \
+    \! -name "Ortheus.py" \
+    \! -name "*citest*.py" \
+    \! -path "*/citest/*" -print0 |
+    xargs -0 mypy --config-file pyproject.toml --namespace-packages --explicit-package-bases
+
+  # Return the exit status of mypy
+  return $?
```

Comment:
Could this be removed so that the return value of the function is the exit status of `mypy`?
```suggestion
```